folium/features.py

```diff
@@ -193,22 +195,41 @@ def __init__(self, data):
         >>> # Providing string.
         >>> GeoJson(open('foo.json').read())
 
+        >>> # Providing a style_function that put all states in green, but Alabama in blue.
+        >>> style_function=lambda x: {'fillColor': '#0000ff' if x['properties']['name']=='Alabama' else '#00ff00'}
+        >>> GeoJson(geojson, style_function=style_function)
         """
         super(GeoJson, self).__init__()
         self._name = 'GeoJson'
         if 'read' in dir(data):
-            self.data = data.read()
+            self.data = json.load(data)
         elif type(data) is dict:
-            self.data = json.dumps(data)
+            self.data = json.loads(json.dumps(data)) # a lazy way of doing deep copy.
```

Comment:
If you have reasons to believe you have nested dicts and you want to do deep copy then just use `copy.deepcopy()`. More Pythonic, no need for comments. :)

By the way, why are you copy the dict? I'd be ok with the map changing after I modify a dict I passed to it. It'd actually surprise me less.